rero_ils/modules/templates/schemas/json.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
+# Copyright (C) 2019-2022 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Schema for JSON representation of `Template` resources."""
+from functools import partial
+
+from invenio_records_rest.schemas import StrictKeysMixin
+from invenio_records_rest.schemas.fields import GenFunction, SanitizedUnicode
+from marshmallow import ValidationError, fields, validates, validates_schema
+from marshmallow.validate import OneOf
+
+from rero_ils.modules.commons.schemas import http_applicable_method
+from rero_ils.modules.patrons.api import current_librarian
+from rero_ils.modules.serializers.base import schema_from_context
+from rero_ils.modules.users.models import UserRole
+
+from ..api import Template
+from ..models import TemplateVisibility
+
+schema_from_template = partial(schema_from_context, schema=Template.schema)
+
+
+class TemplateMetadataSchemaV1(StrictKeysMixin):
+    """Marshmallow schema for the `Template` metadata."""
+
+    pid = SanitizedUnicode()
+    template_type = SanitizedUnicode(required=True)
+    name = SanitizedUnicode(required=True)
+    description = SanitizedUnicode()
+    visibility = SanitizedUnicode(
+        required=True,
+        validate=OneOf([TemplateVisibility.PUBLIC, TemplateVisibility.PRIVATE])
+    )
+    data = fields.Dict()
+    creator = fields.Dict()
+    organisation = fields.Dict()
+    schema = GenFunction(
+        load_only=True,
+        attribute="$schema",
+        data_key="$schema",
+        deserialize=schema_from_template
+    )
+
+    @validates('visibility')
+    @http_applicable_method('POST')
+    def validate_visibility(self, data, **kwargs):
+        """Validate the visibility field through REST API request.
+
+        Through the API, a template could only be created with `private`
+        visibility. To set a template in `public` visibility, an authorized
+        user must update an existing template.
+
+        :param data: the visibility field value.
+        :param kwargs: any additional named arguments.
+        :raises ValidationError: if error has detected on visibility attribute
+        """
+        if data == TemplateVisibility.PUBLIC:
+            raise ValidationError(
+                'Template can be created only with `private` visibility')
```

Comment:
Do we need to do translations?

Reply:
I add this string for babel extraction. 
Personally, It's always a dilemma for me to know where the translation will be done.